Description: Update to R 3.5.1 (incomplete)
 There is still 1 (+1) failing doctest, see the below bug report
 for a detailed description of why it is complex to fix.
Author: Ximin Luo <infinity0@debian.org>
Bug: https://trac.sagemath.org/ticket/26185
--- a/sage/src/sage/interfaces/r.py
+++ b/sage/src/sage/interfaces/r.py
@@ -1784,7 +1784,7 @@
         # [[However, using a file for output often isn't necessary
         # since pipe buffering works pretty well for output.
         # That said, benchmark this.  -- William Stein]]
-        exp = P.eval('dput(%s)'%self.name())
+        exp = P.eval('dput(%s, control=c("keepNA", "keepInteger", "showAttributes"))'%self.name())
 
         # Preprocess expression
         # example what this could be:
@@ -1822,11 +1822,11 @@
 
         # Change 'structure' to '_r_structure'
         # TODO: check that we are outside of quotes ""
-        exp = re.sub(r' structure\(', ' _r_structure(', exp)
-        exp = re.sub(r'^structure\(', '_r_structure(', exp)  # special case
+        exp = re.sub(r'\bstructure\(', '_r_structure(', exp)
+        exp = re.sub(r'^structure\(', '_r_structure(', exp) #special case
 
         # Change 'list' to '_r_list'
-        exp = re.sub(r' list\(', ' _r_list(', exp)
+        exp = re.sub(r'\blist\(', '_r_list(', exp)
         exp = re.sub(r'\(list\(', '(_r_list(', exp)
 
         # Change 'a:b' to 'range(a,b+1)'
